clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperation/SerializeApplicationAtomicOperation.groovy

```diff
@@ -0,0 +1,645 @@
+import com.google.api.services.compute.model.AttachedDiskInitializeParams
+import com.google.api.services.compute.model.AutoscalingPolicy
+import com.google.api.services.compute.model.InstanceProperties
+import com.google.api.services.compute.model.InstanceTemplate
+import com.google.api.services.compute.model.Metadata
+import com.google.api.services.compute.model.NetworkInterface
+import com.google.api.services.compute.model.Scheduling
+import com.google.api.services.compute.model.ServiceAccount
+import com.google.api.services.compute.model.Tags
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.google.deploy.description.SerializeApplicationDescription.SerializeApplicationDescription
+import com.netflix.spinnaker.clouddriver.google.model.GoogleCluster
+import com.netflix.spinnaker.clouddriver.google.model.GoogleHealthCheck
+import com.netflix.spinnaker.clouddriver.google.model.GoogleLoadBalancer
+import com.netflix.spinnaker.clouddriver.google.model.GoogleSecurityGroup
+import com.netflix.spinnaker.clouddriver.google.model.GoogleServerGroup
+import com.netflix.spinnaker.clouddriver.google.provider.view.GoogleClusterProvider
+import com.netflix.spinnaker.clouddriver.google.provider.view.GoogleLoadBalancerProvider
+import com.netflix.spinnaker.clouddriver.google.provider.view.GoogleSecurityGroupProvider
+import com.netflix.spinnaker.clouddriver.model.securitygroups.Rule
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import groovy.json.JsonBuilder
+import org.springframework.beans.factory.annotation.Autowired
+
+class SerializeApplicationAtomicOperation implements AtomicOperation<Void> {
+
+  private static final String BASE_PHASE = "SERIALZIE_APPLICATION"
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  private final description
+  private final String applicationName
+  private final String accountName
+
+  private int numInstanceGroupManagers = 0
+  private int numInstanceTemplates = 0
+  private int numTargetPools = 0
+  private int numForwardingRules = 0
+  private int numHealthChecks = 0
+  private int numAutoscalers = 0
+  private int numFirewalls = 0
+
+  @Autowired
+  GoogleClusterProvider googleClusterProvider
+
+  @Autowired
+  GoogleLoadBalancerProvider googleLoadBalancerProvider
+
+  @Autowired
+  GoogleSecurityGroupProvider googleSecurityGroupProvider
+
+  SerializeApplicationAtomicOperation(SerializeApplicationDescription description) {
+    this.description = description
+    this.applicationName = description.applicationName
+    this.accountName = description.accountName
+  }
+
+
+  /* curl -X POST -H "Content-Type: application/json" -d '[ { "serializeApplication": { "applicationName": "codelab", "credentials": "my-google-account" }} ]' localhost:7002/gce/ops */
+  @Override
+  Void operate(List priorOutputs) {
```

Comment:
You should emit status messages as the operation proceeds. This operation has quite a few status messages that can serve as examples: https://github.com/spinnaker/clouddriver/blob/master/clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleLoadBalancerAtomicOperation.groovy

Search that class for `updateStatus`.

When the requestor of the operation queries for task status, they will be returned the accumulating status messages (and any exceptions).